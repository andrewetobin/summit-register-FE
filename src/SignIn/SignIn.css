.SignIn {
  align-items: center;
  background: url("../utilities/Images/mountains-2.jpg") no-repeat center
  rgba(165, 165, 165, 0.68);
  background-size: auto 100%;
  display: flex;
  justify-content: center;
}

