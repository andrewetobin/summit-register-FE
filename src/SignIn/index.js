import React, { Component } from "react";
import firebase from "firebase";
import PropTypes from "prop-types";
import StyledFirebaseAuth from "react-firebaseui/StyledFirebaseAuth";
import "./SignIn.css";

firebase.initializeApp({
  apiKey: "AIzaSyCOPftNFUOFLFm0alfd8hswAh9kF-wojsc",
  authDomain: "summit-register-dad45.firebaseapp.com"
});

class SignIn extends Component {
  constructor() {
    super();
    this.state = {
      isSignedIn: false
    };
  }

  uiConfig = {
    signInFlow: "popup",
    signInOptions: [firebase.auth.GoogleAuthProvider.PROVIDER_ID],
    callbacks: {
      signInSuccessWithAuthResult: () => false
    }
  };

  componentDidMount = () => {
    firebase.auth().onAuthStateChanged(user => {
      this.setState({ isSignedIn: !!user });
      if (user) {
        this.props.validateSignIn(user.I);
      }
    });
  };

  render() {
    return (
      <div className="SignIn">
        <StyledFirebaseAuth
          uiConfig={this.uiConfig}
          firebaseAuth={firebase.auth()}
        />
      </div>
    );
  }
}

SignIn.propTypes = {
  handleBackButton: PropTypes.func,
  validateSignIn: PropTypes.func
};

export default SignIn;

//===This code gives you back the Token for the current User to send to the backend when we get to the iteration ===//

// console.log(firebase.auth().onAuthStateChanged(function(user) {
//   if (user) {
//     user.getIdToken().then(function(data) {
//       console.log(data)
//     });
//   }
// }))
