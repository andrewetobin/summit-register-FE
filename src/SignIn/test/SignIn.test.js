import React from "react";
import ReactDOM from "react-dom";
import SignIn from "../index";
import { shallow } from "enzyme";

describe("Info", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SignIn />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<SignIn />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
