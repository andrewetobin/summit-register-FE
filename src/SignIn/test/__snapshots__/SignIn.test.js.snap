// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <e
        firebaseAuth={
          Object {
            "apiKey": "AIzaSyCOPftNFUOFLFm0alfd8hswAh9kF-wojsc",
            "appName": "[DEFAULT]",
            "authDomain": "summit-register-dad45.firebaseapp.com",
            "currentUser": null,
          }
        }
        uiConfig={
          Object {
            "callbacks": Object {
              "signInSuccessWithAuthResult": [Function],
            },
            "signInFlow": "popup",
            "signInOptions": Array [
              "google.com",
              "password",
            ],
          }
        }
      />,
      "className": "SignIn",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "firebaseAuth": Object {
          "apiKey": "AIzaSyCOPftNFUOFLFm0alfd8hswAh9kF-wojsc",
          "appName": "[DEFAULT]",
          "authDomain": "summit-register-dad45.firebaseapp.com",
          "currentUser": null,
        },
        "uiConfig": Object {
          "callbacks": Object {
            "signInSuccessWithAuthResult": [Function],
          },
          "signInFlow": "popup",
          "signInOptions": Array [
            "google.com",
            "password",
          ],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <e
          firebaseAuth={
            Object {
              "apiKey": "AIzaSyCOPftNFUOFLFm0alfd8hswAh9kF-wojsc",
              "appName": "[DEFAULT]",
              "authDomain": "summit-register-dad45.firebaseapp.com",
              "currentUser": null,
            }
          }
          uiConfig={
            Object {
              "callbacks": Object {
                "signInSuccessWithAuthResult": [Function],
              },
              "signInFlow": "popup",
              "signInOptions": Array [
                "google.com",
                "password",
              ],
            }
          }
        />,
        "className": "SignIn",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "firebaseAuth": Object {
            "apiKey": "AIzaSyCOPftNFUOFLFm0alfd8hswAh9kF-wojsc",
            "appName": "[DEFAULT]",
            "authDomain": "summit-register-dad45.firebaseapp.com",
            "currentUser": null,
          },
          "uiConfig": Object {
            "callbacks": Object {
              "signInSuccessWithAuthResult": [Function],
            },
            "signInFlow": "popup",
            "signInOptions": Array [
              "google.com",
              "password",
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
